src/promptflow-devkit/promptflow/_sdk/_tracing.py

```diff
@@ -640,6 +637,15 @@ def _try_write_trace_to_cosmosdb(
         logger.info(f"Start writing trace to cosmosdb, total spans count: {len(all_spans)}.")
         start_time = datetime.now()
 
+        # use Azure CLI credential init as default get credential function
+        # outside this function, or say before this line, we might be in an environment with only promptflow-devkit
+        # we cannot import `AzureCliCredential` from `azure-identity`, so it might be None
+        # for other usages like runtime, or PRS, they should pass the function to get credential
+        if get_credential is None:
+            from azure.identity import AzureCliCredential
+
+            get_credential = AzureCliCredential
```

Comment:
nit: Is check possible to use `_is_azure_ext_installed` check about azure dependency exist and pass in as get_credential?
Still not sure if that's a good practice to use `AzureCliCredential` as default value.

Reply:
updated one version, will directly pass an exception class when azure extension is not installed - this can make no default value logic inside that parse function.